package/MDAnalysis/analysis/dssp/dssp.py

```diff
@@ -0,0 +1,396 @@
+    Doi("10.1002/bip.360221211"),
+    description="DSSP algorithm description",
+    path="MDAnalysis.analysis.dssp",
+    cite_module=True,
+)
+
+del Doi
+
+
+try:  # pragma: no cover
+    from pydssp.pydssp_numpy import (
+        assign,
+        _get_hydrogen_atom_position,
+    )
+
+    HAS_PYDSSP = True
+
+except ModuleNotFoundError:
+    HAS_PYDSSP = False
+    from .pydssp_numpy import (
+        assign,
+        _get_hydrogen_atom_position,
+    )
+
+
+class DSSP(AnalysisBase):
+    """Assign secondary structure using the DSSP algorithm.
+
+    Analyze a selection containing a protein and assign secondary structure
+    using the Kabsch-Sander algorithm :footcite:p:`Kabsch1983`. Only a subset
+    of secondary structure categories are implemented:
+
+    - 'H' represents a generic helix (α-helix, π-helix or :math:`3_{10}` helix)
+    - 'E' represents 'extended strand', participating in beta-ladder (parallel
+      or antiparallel)
+    - '-' represents unordered part
+
+    The implementation was taken from the pydssp package (v. 0.9.0)
+    https://github.com/ShintaroMinami/PyDSSP by Shintaro Minami.
+
+    .. Warning::
+        For the DSSP to work properly, your atoms (either as ``Universe``
+        or ``AtomGroup``) must have only one atom named "H" per residue,
+        that corresponds to N-bound hydrogen in a peptide bond (or zero,
+        if the residue is proline). It is the default case for most of
+        the modern syntaxes, but if you encounter an error or unusual results
+        during your run, please report an issue on MDAnalysis
+        [github](https://github.com/MDAnalysis/mdanalysis/issues).
+
+    Parameters
+    ----------
+    atoms : Union[Universe, AtomGroup]
+        input Universe or AtomGroup. In both cases, only protein residues will
+        be chosen prior to the analysis via `select_atoms('protein')`.
+        Heavy atoms of the protein are then selected by name
+        ("N", "CA", "C", "O O1 OT1"), and hydrogens are selected by name "H".
+    guess_hydrogens : bool, optional
+        whether you want to guess hydrogens positions, by default ``True``.
+        Guessing is made assuming perfect 120 degrees for all bonds that N
+        atom makes, and a N-H bond length of 1.01 A.
+        If ``guess_hydrogens`` is False, hydrogen atom positions on N atoms
+        will be parsed from the trajectory, except for the "hydrogen" atom
+        positions on PRO residues, and an N-terminal residue.
+    heavyatom_names : tuple[str], default ("N", "CA", "C", "O O1 OT1")
+        selection names that will be used to select "N", "CA", "C" and "O"
+        atom coordinates for the secondary structure determination. The last
+        string contains multiple values for "O" to account for C-term residues.
+    hydrogen_name : str, default "H HN"
+        selection string that will be used to select N-bound hydrogen in every
+        residue. Default string contains both "H" and "HN" simultaneously, since
```

Comment:
I don't understand the remark. Under which circumstances is it correct to use "H" and under which "HN"? 

When does "HN" occur?

Reply:
I believe that different syntaxes give different values. For example, in the NhaA that you provided link to, N-terminal SER has `HT1 HT2 HT3` as hydrogens at N-term (I believe it stands for "Hydrogen Terminal ..."), and then `HN` all over the sequence, instead of simple `H`.

Here I merged all the syntaxes that I've met before, hoping that I won't meet simultaneously any of these two selections on one residue. It could be implemented as `init`-check, for example.

Reply:
Ok, now I understand better: These are different names that appear in different force fields/naming standards and you are trying to match one.

To make the *intent* clearer, I'd replace the sentence "Default string contains both "H" and "HN" simultaneously, since they aren't used together in topologies." with something like

```reST
This selection should only select a single hydrogen atom in each residue
(except proline), namely the one bound to the backbone nitrogen. 

.. Note::
   To work with different hydrogen-naming conventions by default, the default
   selection is broad but if hydrogens are incorrectly selected (e.g., a
   :exc:`ValueError` is raised you must customize `hydrogen_name` for your 
   specific case.

```